framework/deproxy_server.py

```diff
@@ -41,13 +43,16 @@ def initiate_send(self):
             self.server.segment_size if self.server.segment_size else deproxy.MAX_MESSAGE_SIZE
         )
 
-        sent = self.socket.send(self.out_buffer[:segment_size])
+        resp = self.response_buffer[self.responses_done]
+        sent = self.socket.send(resp[:segment_size])
+
         if sent < 0:
             return
-        self.out_buffer = self.out_buffer[sent:]
+        self.response_buffer[self.responses_done] = resp[sent:]
 
         self.last_segment_time = time.time()
-        self.responses_done += 1
+        if self.response_buffer[self.responses_done] == b"":
```

Comment:
Just style. You can use [Boolean Operations](https://docs.python.org/3/library/stdtypes.html#truth-value-testing) here.

Reply:
I prefer to use `==` because it looks clearer for me.